recipes/wasmtime/all/conanfile.py

```diff
@@ -60,16 +65,22 @@ def validate(self):
             self.output.warn(msg)
 
         try:
-            self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)]
+            self.conan_data["sources"][self.version][self._sources_os_key][str(self.settings.arch)]
         except KeyError:
-            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os are not available")
 
         if (self.settings.compiler, self.settings.os) == ("gcc", "Windows") and self.options.shared:
             # FIXME: https://github.com/bytecodealliance/wasmtime/issues/3168
             raise ConanInvalidConfiguration("Shared mingw is currently not possible")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)], destination=self.source_folder, strip_root=True)
+    def package_id(self):
+        del self.info.settings.compiler.version
+        if self.settings.compiler == "clang":
+            self.info.settings.compiler = "gcc"
+
+    def build(self):
```

Comment:
I remember I have made the same way in first commit but the received comment that it is wrong for receipt for `conan-center-index` ...
All repeats )))

Let's see what comments will be ...

Reply:
Indeed, your initial wasmtime pr correctly downloaded the binaries in the `build` method in the first commit.
But I can't find the suggestion that asked you to move it to `source`.

Reply:
@madebr I do not remember who exactly asked me to do this, but it was not my own decision ...

Reply:
@madebr I have found who changed `build` -> `source` https://github.com/conan-io/conan-center-index/pull/6083#discussion_r663509985

Reply:
Darn it! Sorry!

Reply:
```suggestion
    def build(self):
        # This is packaging binaries so the download needs to be in build
```